front/lib/models/assistant/actions/retrieval.ts

```diff
@@ -185,18 +192,32 @@ AgentDataSourceConfiguration.init(
 AgentRetrievalConfiguration.hasOne(AgentConfiguration, {
   foreignKey: { name: "retrievalConfigurationId", allowNull: true }, // null = no retrieval action set for this Agent
 });
+AgentConfiguration.belongsTo(AgentRetrievalConfiguration, {
+  as: "retrievalConfiguration",
+  foreignKey: { name: "retrievalConfigurationId", allowNull: true },
+});
 
 // Retrieval config <> Data source config
 AgentRetrievalConfiguration.hasMany(AgentDataSourceConfiguration, {
+  as: "dataSourceConfigurations",
   foreignKey: { name: "retrievalConfigurationId", allowNull: false },
   onDelete: "CASCADE",
 });
+AgentDataSourceConfiguration.belongsTo(AgentRetrievalConfiguration, {
+  as: "dataSourceConfigurations",
+  foreignKey: "retrievalConfigurationId",
+});
 
 // Data source config <> Data source
 DataSource.hasMany(AgentDataSourceConfiguration, {
+  as: "ds",
   foreignKey: { name: "dataSourceId", allowNull: false },
   onDelete: "CASCADE",
 });
+AgentDataSourceConfiguration.belongsTo(DataSource, {
```

Comment:
Why do we need this?

Reply:
This one is incorrect as it declares a One-To-One which contradicts the One-To-Many above. the `as` can be added in the `hasMany` above and this declaration can be removed